pytest_jupyterhub/jupyterhub_spawners.py

```diff
@@ -5,9 +5,18 @@
 
 import pytest
 from jupyterhub.spawner import Spawner
+from traitlets.config import Config
 
 
 @pytest.fixture
 def spawner_class():
     """Allows tests to setup the Spawner class that will be used by the hub."""
     return Spawner
+
+
+@pytest.fixture
+def jupyterhub_spawner_config(spawner_class):
+    """Configures JupyterHub to use the Spawner class"""
+    c = Config()
+    c.JupyterHub.spawner_class = spawner_class
+    return c
```

Comment:
Ideally, when you know the arguments and you don't need to process them in any way, you would call the object constructor with those arguments in order to initialize the variables with these args, instead of first creating the object and then set its variables.

Example:
```python
   spawner_class_config = { 
       "JupyterHub": {
           "spawner_class": spawner_class
         }
     }

    return Config(spawner_class_config)
```

Reply:
so what `Config(spawner_class_config)` is, in your example, is the object constructor with the `spawner_class` argument already initialized?

Reply:
Something like this. This is the [source code of `traitlets.Config`](https://docs.python.org/3/library/stdtypes.html#dict) if your're curious of what actually happens when `Config(spawner_class_config)` is called.  `traitlets.Config` it's pretty much a [`dict`](https://docs.python.org/3/library/stdtypes.html#dict) with some extra capabilities.